guardian/editorial-collaboration
# Save steps to database

## What does this change?
This pr adds the functionality to save steps sent to the `/documents/:id/steps` endpoint to be saved to the postgres database. It uses local config when running locally, or SecretsManager to retrieve database config in CODE and PROD.

## How to test
Testing locally: (re)run `./scripts/setup` (ideally deleting any Dockerised database you may already have). Your local db Docker container should be set-up correctly using the `.env.example` file. Start the app, start local Composer and enable the feature switch to send steps to the collab server. As you type in an article, you should see the steps appearing in your local database.

Testing in CODE: We need to run the migration on the CODE database to set up the tables. Once that is complete it will be possible to test in CODE using a similar process as local.

## Next steps
TODO: Update the `secretName` in the CDK to be fully parameretised (ie:  `/${stage}/${stack}/${appName}/db`). I resisted doing this as part of this pr as it will likely be another destructive action requiring db admin perms for the Composer account and I wanted to get these change in beforehand to unblock load testing the service.

## How can we measure success?
We should be able to write steps to the database and proceed with load testing the collaboration service.
